core/iwasm/libraries/lib-socket/test/tcp.c

```diff
@@ -0,0 +1,90 @@
+#include <unistd.h>
+#include <string.h>
+#include "assert.h"
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <arpa/inet.h>
+#endif
+#include <pthread.h>
+
+#define PORT 8989
+#define IP "127.0.0.1"
+
+struct init_socket {
+    struct sockaddr_in addr;
+    int sock;
+};
+
+struct init_socket
+init_socket_addr()
+{
+    int sock = socket(AF_INET, SOCK_STREAM, 0);
+    assert(sock > 0);
+
+    struct sockaddr_in addr;
+    memset(&addr, '\0', sizeof(addr));
+    addr.sin_family = AF_INET;
+    addr.sin_port = PORT;
+    addr.sin_addr.s_addr = inet_addr(IP);
+    return (struct init_socket){ addr, sock };
+}
+
+void *
+server()
+{
+    struct init_socket init_server_sock = init_socket_addr();
+    struct sockaddr_in server_addr = init_server_sock.addr, client_addr;
+    int server_sock = init_server_sock.sock, client_sock;
+    socklen_t addr_size;
+    char buffer[1024];
+
+    int n =
+        bind(server_sock, (struct sockaddr *)&server_addr, sizeof(server_addr));
+    assert(n >= 0);
+
+    listen(server_sock, 1);
+    addr_size = sizeof(client_addr);
+    client_sock =
+        accept(server_sock, (struct sockaddr *)&client_addr, &addr_size);
+    assert(client_sock >= 0);
+
+    bzero(buffer, 1024);
+    strcpy(buffer, "Message from server.");
+    sendto(client_sock, buffer, strlen(buffer), 0,
+           (struct sockaddr *)&client_addr, addr_size);
+
+    assert(close(client_sock) >= 0);
+    return NULL;
+}
+
+void *
+client()
+{
+    struct init_socket init_client_sock = init_socket_addr();
+    int sock = init_client_sock.sock;
+    struct sockaddr_in addr = init_client_sock.addr;
+    socklen_t addr_size;
+    char buffer[1024];
+    assert(connect(sock, (struct sockaddr *)&addr, sizeof(addr)) >= 0);
+
+    bzero(buffer, 1024);
+    recv(sock, buffer, sizeof(buffer), 0);
+    assert(strcmp(buffer, "Message from server.") == 0);
+    assert(close(sock) >= 0);
+    return NULL;
+}
+
+int
+main()
+{
+    pthread_t thread1,
+        thread2; // one thread runs the server, other runs the client
```

Comment:
maybe we can use more descriptive names instead of the comment? e.g. `server_thread` and `client_thread`?